x/savings/keeper/deposit.go

```diff
@@ -21,13 +21,20 @@ func (k Keeper) Deposit(ctx sdk.Context, depositor sdk.AccAddress, coins sdk.Coi
 	}
 
 	currDeposit, foundDeposit := k.GetDeposit(ctx, depositor)
-	amount := coins
+
+	deposit := types.NewDeposit(depositor, coins)
 	if foundDeposit {
-		amount = amount.Add(currDeposit.Amount...)
+		deposit.Amount = deposit.Amount.Add(currDeposit.Amount...)
+		k.hooks.BeforeSavingsDepositModified(ctx, deposit, getDenoms(coins))
```

Comment:
pretty sure `incomingDenoms` should be defined as only the new denoms. As currently written, if I have an existing `ukava` deposit to the savings module and add another `ukava` deposit, getDenoms will return `ukava` to the `incomingDemons` argument. What we want is the set difference between `coins` and `deposit.Amount`.